### ISSUE TITLE
As a user, I want to search across interventions so I can see the kinds of marks and other interventions made in Derrida's Library.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/derrida-django#119]

**Comment by rlskoeser** [comment]
@kevinglover I have a first pass at intervention search working.  I adapted your list collection item and created an intervention list item template - looks nearly ok except for the page images.  Would you please take a look?

**Comment by rlskoeser** [comment]
Page images look fine now that I've made some updates to the image handling.

**Comment by araiffe** [comment]
@rlskoeser Rebecca, could you also provide me with instructions on how to test for this one? Thank you!

**Comment by rlskoeser** [comment]
@araiffe the only entry point is from the search (unless you happen to know the url and go there directly) - either the cross item search or using the search and selecting interventions.  But if you can't get the search form to work, that's not very helpful!  We'll figure out what's going on with that and get it resolved.

**Comment by araiffe** [comment]
@rlskoeser @xinyil I am not sure I did this right, but I selected "Interventions" in the search menu and typed "arrow" and I'm getting something that looks like this: is this correct?
![screen shot 2017-10-24 at 11 35 56](https://user-images.githubusercontent.com/26753387/31952859-99b2a8aa-b8af-11e7-8924-13f2d88b2525.png)

**Comment by rlskoeser** [comment]
@araiffe sorry for the confusion - yes, that's correct.  The facets haven't been integrated into the drop-down filter display yet like the other views, but the filtering logic _should_ be working correctly.  Test for search filter functionality only, don't worry about how the form is displaying (we'll get it styled soon.)

**Comment by araiffe** [comment]
@rlskoeser @kac3 also…what are the search terms that are operable in this field? I typed "Rousseau" for example (i.e. to see what interventions Derrida made on Rousseau's works) and nothing comes up. Can  a user only search "line", "arrow" etc.?

**Comment by rlskoeser** [comment]
@araiffe great question.  I just checked the code, and I'm currently only indexing fields directly on the intervention in the default search (text, anchor text, languages, author of the intervention) - but I agree that searching by author or title of the work instance that has the annotation makes a lot of sense!  Would author + title be enough?  Looking at the other book-specific fields, they don't seem as relevant to searching on interventions, but let me know.

**Comment by araiffe** [comment]
@rlskoeser I think this is a question for @kac3 
Katie, could you take a look at this and share your thoughts? It seems directly relevant to also look at issue #121, which is related to this one, as to how searching through interventions would work.

**Comment by kac3** [comment]
@rlskoeser @araiffe  This is tricky.  

On the one hand, I definitely see the value of being able to search by author/title in this field.  On the other hand, I see the value of limiting it to text/anchor text so that when, for example, a user searches for Rousseau she finds only interventions that mention the name Rousseau rather than every intervention in every Rousseau text -- especially when we have the functionality of #121 that allows a user to do the latter in a another way.  

In other words, it seems to me that adding author/title fields seems redundant BUT nevertheless useful and perhaps more intuitive (as it seems to have been for you, Alex).  What we lose by adding it is the ability to search only the text/anchor text fields.  I'm having a hard time anticipating which will be more crucial for users.

@rlskoeser What do you think?

**Comment by rlskoeser** [comment]
@kac3 @araiffe yes, I see how this is problematic.  We want people to be able to find content easily, but we don't want to obscure or make it harder to find what is really the more significant content.

If those are our only two options, I would say we should probably leave author and title out of the full text search and find a good way to direct people to use the filters.  However, I did a little experimenting and I can offer a third option (hopefully not opening a can of worms or adding too much more complication).

I can modify the intervention search so it also searches on author and title in addition to the text and anchor fields, but in a way that scores matches on the author and title as less relevant than matches elsewhere.  This _should_ mean that you would see interventions where your search term matches text/anchor text/etc first, and then interventions from a work with a matching author or title.  I happened to discover "zeit" as a useful search term that appears both in intervention content and a title, and it looks to me like this is working.  If you want to experiment with this, I can check in the code and put it up on the test site for you to test.  (At least one limitation in this approach is that all terms have to match in one or the other field - a term from the intervention content and a term from a title won't give you a result.)

It did occur to me along the way that we actually don't have an option to "sort by relevance" anywhere on the site, and I want to talk to Xinyi about that to see if there's something I'm missing.  We only get a relevance score when there is a keyword search, but it seems to me that sorting by relevance is kind of important.  (Although I know it can be vague and confusing to users, since it's not always clear what "relevance" means.)  How do you two feel about this?

If you do want to test this, I could temporarily display the relevance score to help you get an idea of how the search boosting is working.

Unfortunately we don't have much time to customize the indexing at this point - so if you think anything I've proposed is going to be incomplete, cause problems, or require a lot of refining and testing, we should postpone it.

**Comment by xinyil** [comment]
I feel the limitation of the alternative approach rebecca mentioned (all terms have to match in one or the other field - a term from the intervention content and a term from a title won't give you a result) might create frustration and it’s difficult to explain to user why there’s no results, and what’s the proper way to search. 

If we limit the search to only text and anchor text, we can change the instructional message in the text field, so when the drop down menu is selected to intervention, the hint says “search text and anchor text” to give the correct anticipation.   


We can definitely add sorting by relevance in the sorting option part.

**Comment by rlskoeser** [comment]
I agree with Xinyi about the potential frustrations and think it's better to keep it simple.

I like the sort by relevance and think it would be valuable, but I'm not comfortable adding any more features at this point since we're trying to close things out.  Let's add it to a wishlist somewhere for a later version.
